cms/djangoapps/contentstore/tests/test_courseware_index.py

```diff
@@ -551,15 +571,227 @@ def setUp(self):
+        self.html_unit2 = ItemFactory.create(
+            parent_location=self.library.location,
+            category="html",
+            display_name="Html Content 2",
+            modulestore=store,
+            publish_item=False,
+        )
+
+    INDEX_NAME = LibrarySearchIndexer.INDEX_NAME
+
+    def _get_default_search(self):
+        """ Returns field_dictionary for default search """
+        return {"library": unicode(self.library.location.library_key.replace(version_guid=None, branch=None))}
+
+    def reindex_library(self, store):
+        """ kick off complete reindex of the course """
+        return LibrarySearchIndexer.do_library_reindex(store, self.library.location.library_key)
+
+    def _get_contents(self, response):
+        """ Extracts contents from search response """
+        return [item['data']['content'] for item in response['results']]
+
+    def index_recent_changes(self, store, since_time):
+        """ index course using recent changes """
+        trigger_time = datetime.now(UTC)
+        return LibrarySearchIndexer.index(
+            store,
+            self.library.id,
+            triggered_at=trigger_time,
+            reindex_age=(trigger_time - since_time)
+        )
+
+    def _test_indexing_library(self, store):
+        """ indexing course tests """
+        self.reindex_library(store)
+        response = self.search()
+        self.assertEqual(response["total"], 2)
+
+        added_to_index = self.reindex_library(store)
+        self.assertEqual(added_to_index, 2)
+        response = self.search()
+        self.assertEqual(response["total"], 2)
+
+    def _test_creating_item(self, store):
+        """ test updating an item """
+        self.reindex_library(store)
+        response = self.search()
+        self.assertEqual(response["total"], 2)
+
+        # updating a library item causes immediate reindexing
+        data = "Some data"
+        ItemFactory.create(
+            parent_location=self.library.location,
+            category="html",
+            display_name="Html Content 3",
+            data=data,
+            modulestore=store,
+            publish_item=False,
+        )
+
+        self.reindex_library(store)
+        response = self.search()
+        self.assertEqual(response["total"], 3)
+        html_contents = [cont['html_content'] for cont in self._get_contents(response)]
+        self.assertIn(data, html_contents)
+
+    def _test_updating_item(self, store):
+        """ test updating an item """
+        self.reindex_library(store)
+        response = self.search()
+        self.assertEqual(response["total"], 2)
+
+        # updating a library item causes immediate reindexing
+        new_data = "I'm new data"
+        self.html_unit1.data = new_data
+        self.update_item(store, self.html_unit1)
+        self.reindex_library(store)
+        response = self.search()
+        # TODO: MockSearchEngine never updates existing item: returns 3 items here - uncomment when it's fixed
```

Comment:
@Kelketek thank you for reminding me - I totally forgot about this TODO. [SOL-731](https://openedx.atlassian.net/browse/SOL-731) at your service. 

@martynjames it might make sense to prioritize update it, but I'm not sure if it belongs to SOL project and who's responsible for reviewing and prioritizing search-related tasks. Can you please take a look and review/redirect the task?